cli/azd/pkg/experimentation/client_test.go

```diff
@@ -0,0 +1,98 @@
+package experimentation
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/azure/azure-dev/cli/azd/test/mocks/mockhttp"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetVariantAssignments(t *testing.T) {
+	endpoint := "https://test-exp-s2s.msedge.net/ab"
+
+	mockHttp := mockhttp.NewMockHttpUtil()
+	mockHttp.When(func(request *http.Request) bool {
+		return request.Method == http.MethodGet && request.URL.String() == endpoint
+	}).RespondFn(
+		func(request *http.Request) (*http.Response, error) {
+			// nolint: staticcheck
+			require.ElementsMatch(t, request.Header["X-ExP-Parameters"], []string{"key1=value1", "key2=value2"})
+			// nolint: staticcheck
+			require.Equal(t, request.Header["X-ExP-RequiredVariants"], []string{"variant1", "variant2"})
+			// nolint: staticcheck
+			require.Equal(t, request.Header["X-ExP-BlockedVariants"], []string{"variant3", "variant4"})
+			// nolint: staticcheck
+			require.Equal(t, request.Header["X-ExP-AssignmentScopes"], []string(nil))
+
+			res := treatmentAssignmentResponse{
+				FlightingVersion:  1,
+				AssignmentContext: "context:393182",
+			}
+
+			jsonBytes, _ := json.Marshal(res)
+
+			return &http.Response{
+				Request:    request,
+				StatusCode: http.StatusOK,
+				Header:     http.Header{},
+				Body:       io.NopCloser(bytes.NewBuffer(jsonBytes)),
+			}, nil
+		},
+	)
+	client, err := newTasClient("https://test-exp-s2s.msedge.net/ab", &azcore.ClientOptions{
+		Transport: mockHttp,
+	})
+	require.NoError(t, err)
+
+	request := variantAssignmentRequest{
+		Parameters: map[string]string{
+			"key1": "value1",
+			"key2": "value2",
+		},
+		AssignmentScopes: []string{""},
+		RequiredVariants: []string{"variant1", "variant2"},
+		BlockedVariants:  []string{"variant3", "variant4"},
+	}
+	resp, err := client.GetVariantAssignments(context.Background(), &request)
+	require.NoError(t, err)
+	require.Equal(t, "context:393182", resp.AssignmentContext)
+	require.Equal(t, int64(1), resp.FlightingVersion)
+}
+
+func TestGetEscapedParameterStrings(t *testing.T) {
```

Comment:
nit: maybe update the tests to be inline with our assertions, and switch to table-style input tests.